src/WebJobs.Extensions.DurableTask/DurableHttpResponse.cs

```diff
@@ -33,47 +30,27 @@ public class DurableHttpResponse
             string content = null)
         {
             this.StatusCode = statusCode;
-            this.Headers = headers ?? new Dictionary<string, StringValues>(StringComparer.OrdinalIgnoreCase);
+            this.Headers = HttpHeadersConverter.CreateCopy(headers);
```

Comment:
My initial reaction is also yes. In this specific case I wanted us to do so because we want to make sure that we're assuming that we have a case-insensitive dictionary, and there's no guarantee that the user would pass us a case-insensitive one. 

Generally speaking, it seems there are two different possibilities of bugs. I assume you're talking about the case where they pass in a dictionary and modify that dictionary in their code afterwards.

* Case 1: The user expects that their dictionary modifications would also modify the state of `DurableHttpResponse`.
* Case 2: The user expects that their dictionary modifications won't affect the state of `DurableHttpResponse` - or they don't think about it at all and just modify the dictionary.

I'm more worried about Case 2 because that's more likely to result in _accidental_ behavior changes. Case 1 seems like a poor/risky assumption on the user's part, so I'm less worried about being defensive about it. If they want to modify the headers afterwards, they should just access the `Headers` property directly.